modules/xml/test/src/smithy4s/xml/XmlCodecSpec.scala

```diff
@@ -621,27 +607,20 @@ object XmlCodecSpec extends SimpleIOSuite {
       .map(result => expect.same(result, expected))
   }
 
-  // Decode document differs from decode content in that the top-level
-  // tag is checked against the ShapeId
-  private def decodeDocument[A: Schema](document: XmlDocument): IO[A] = {
```

Comment:
I couldn't figure out why this was necessary. I believe I've been able to simplify things without losing anything, but might be worth a second opinion.